apps/content-gateway-api/src/app/maintenance/MaintenanceJob.ts

```diff
@@ -0,0 +1,43 @@
+import { ProgramError } from "@banklessdao/util-data";
+import { isLeft } from "fp-ts/lib/Either";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import {
+    AsyncTask,
+    SimpleIntervalJob,
+    SimpleIntervalSchedule,
+} from "toad-scheduler";
+import { Logger } from "tslog";
+import { MaintenanceJobError } from "./errors";
+
+/**
+ * A job designed to perform maintenance tasks on the gateway
+ */
+export interface MaintenanceJob {
+    id: string;
+    run: () => TaskEither<MaintenanceJobError, void>;
+    schedule: SimpleIntervalSchedule;
+}
+
+export type addMaintenanceJob = (
+    job: MaintenanceJob
+) => TaskEither<ProgramError, void>;
+
+export const makeToToadJob =
+    (logger: Logger) =>
+    (job: MaintenanceJob): SimpleIntervalJob => {
+        const task = new AsyncTask(job.id, () => {
+            return job
+                .run()()
+                .then(
+                    (v) => {
+                        if (isLeft(v))
+                            throw v.left.error || new Error("Unknown Cause");
+                        logger.info(`Maintenance Job ${job.id} successful`)
+                    },
+                    (err) => {
+                        logger.warn(`Maintenance Job ${job.id} failed`, err);
+                    }
+                );
+        });
+        return new SimpleIntervalJob(job.schedule, task, job.id);
```

Comment:
This is a great question. Looking into the documentation for Toad Scheduler, I can't really tell. Seems like it's certainly possible it could crash (just like any other piece of software), but I'm not even sure how we'd go about doing that without using Toad Scheduler to check itself (although if we have a separate instance that would add a significant level of robustness). @adam-arold did you do anything when you used Toad Scheduler to account for this possibility?

Reply:
No, I checked the code and it just calls `setTimeout` so in order for this to break `setTimeout` has to be buggy. I don't know if there is an appropriately robust solution for this.